### PR TITLE
add VpnGW skus

### PR DESCRIPTION
**Pull request by Lucretius**
Resolves https://github.com/terraform-providers/terraform-provider-azurerm/issues/2383

Unfortunately, these were not available in the existing network API version, so I bumped to the latest one.  This is my first time bumping a vendored Go project, not sure what the two changes in the `go.sum` file are all about but the existing unit tests pass and the project builds.

### CODE REVIEW

File: azurerm/resource_arm_virtual_network_gateway.go [hunk 1]
```diff
@@ -91,6 +91,9 @@ func resourceArmVirtualNetworkGateway() *schema.Resource {
 					string(network.VirtualNetworkGatewaySkuNameErGw1AZ),
 					string(network.VirtualNetworkGatewaySkuNameErGw2AZ),
 					string(network.VirtualNetworkGatewaySkuNameErGw3AZ),
+					string(network.VirtualNetworkGatewaySkuNameVpnGw1AZ),
+					string(network.VirtualNetworkGatewaySkuNameVpnGw2AZ),
+					string(network.VirtualNetworkGatewaySkuNameVpnGw3AZ),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_virtual_network_gateway.go]: could we [update the documentation to include these new values](https://github.com/terraform-providers/terraform-provider-azurerm/blame/master/website/docs/r/virtual_network_gateway.html.markdown#L138)?

**Comment by Lucretius** [comment]
documentation has been updated with the new values

**Comment by tombuildsstuff** [comment]
@Lucretius I hope you don't mind but so that we can get this merged I've rebased this

**Comment by JamesDLD** [comment]
Hello,

I just tested to use the SKU "VpnGw1AZ" and got the following error message 
`azurerm_virtual_network_gateway.ngw: expected sku to be one of [Basic Standard HighPerformance VpnGw1 VpnGw2 VpnGw3], got VpnGw1AZ`

Through PowerShell the Zonal feature is set on the Public Ip resource object. Isn't there any incomprehension on the Terraform resource **azurerm_virtual_network_gateway**?
See Microsoft documentation : https://docs.microsoft.com/en-us/azure/vpn-gateway/create-zone-redundant-vnet-gateway

Why does the Terraform resource **azurerm_virtual_network_gateway**  has the following sentence (here : https://www.terraform.io/docs/providers/azurerm/r/virtual_network_gateway.html) :
"sku - (Required) Configuration of the size and capacity of the virtual network gateway. Valid options are Basic, Standard, HighPerformance, UltraPerformance, **ErGw1AZ**, **ErGw2AZ**, **ErGw3AZ**, VpnGw1, VpnGw2, VpnGw3, **VpnGw1AZ**, **VpnGw2AZ**, and **VpnGw3AZ** and depend on the type and vpn_type arguments. A PolicyBased gateway only supports the Basic sku. Further, the UltraPerformance sku is only supported by an ExpressRoute gateway."
FYI @tombuildsstuff 

Thank you 
James

**Comment by JamesDLD** [comment]
As discussed on the slack community channel, it appears that the validate function "validateArmVirtualNetworkGatewayRouteBasedVpnSku" doesn't include the SKUs VpnGw1AZ, VpnGw2AZ and VpnGw3AZ for the resource "azurerm_virtual_network_gateway".

**Comment by JamesDLD** [comment]
Did raise a bug here : https://github.com/terraform-providers/terraform-provider-azurerm/issues/3314

**Pull request merged**